### PR TITLE
Implement @optStrategy UDA.

### LINKED ISSUE #1549: Add UDA for per-function optimization settings
Implement support for `optnone`, `optsize`, `minsize` (which I think are all related so perhaps one UDA can be designed for them?).  `optnone` is the most interesting I think. In a PS4 presentation [1] it is mentioned as a great debugging tool when -O0 builds are not an option; makes total sense!

[1] http://llvm.org/devmtg/2013-11/slides/Robinson-PS4Toolchain.pdf

### PR DESCRIPTION
**Pull request by JohanEngelen**
I need this for debugging so implemented it. Any alternative ideas for the name "optStrat"?

Resolves #1549.

druntime change: https://github.com/ldc-developers/druntime/commit/1a718598bc8bf6ffbc68492544df33fe8558b8a4

### CODE REVIEW

**Comment by kinke** [comment]
Thanks, makes a lot of sense. I'd prefer `optStrategy` to make clear we're not talking about optimization **strat**osphere or something. ;)

**Comment by JohanEngelen** [comment]
I changed the name to `optStrategy`.

**Comment by JohanEngelen** [comment]
Fixed merge conflict (because of druntime changes).

ready to merge?

File: gen/uda.cpp [hunk 1]
```diff
@@ -147,6 +148,30 @@ void applyAttrLLVMFastMathFlag(StructLiteralExp *sle, IrFunction *irFunc) {
   }
 }
 
+void applyAttrOptStrat(StructLiteralExp *sle, IrFunction *irFunc) {
```
Comment (kinke) [inline gen/uda.cpp]: OptStrategy here as well?

Comment (JohanEngelen) [inline gen/uda.cpp]: done

**Comment by JohanEngelen** [comment]
thanks!
(now on to the real bug then......)

**Pull request merged**